src/status_im/contexts/communities/events.cljs

```diff
@@ -68,14 +71,19 @@
    (when community-js
      (let [{:keys [token-permissions
                    token-permissions-check joined id]
-            :as   community}      (<-rpc community-js)
-           has-token-permissions? (not (seq token-permissions))]
+            :as   community} (<-rpc community-js)
+           has-channel-perm? (fn [id-perm-tuple]
+                               (let [{:keys [type]} (second id-perm-tuple)]
+                                 (or (= type constants/community-token-permission-can-view-channel)
+                                     (=
+                                      type
+                                      constants/community-token-permission-can-view-and-post-channel))))]
```

Comment:
I tried to add another has-member-perm?
something like `(= type constants/community-token-permission-become-member/admin)`
but code elsewhere relies on `token-permissions-check` to determin if reqest to join is clickable
and `token-permissions-check` relies on `:communities/check-permissions-to-join-community` to set it
@ilmotta

Reply:
It's not clear to me what to do @yqrashawn. For instance, this line on the view is suspicious, probably buggy https://github.com/status-im/status-mobile/blob/5cc027a1f58a35f4b7980612a9080a621d134d6f/src/status_im/contexts/communities/overview/view.cljs#L183

Another place:

https://github.com/status-im/status-mobile/blob/5cc027a1f58a35f4b7980612a9080a621d134d6f/src/status_im/subs/communities.cljs#L311-L313

What I mean is that we should probably create some functions in `data_store/` that are well tested and always refer to them from subscriptions, events and views.

Reply:
I'm not suggesting any changes in your PR, but something for us to consider refactoring soon if possible.